src/commonMain/kotlin/fr/acinq/lightning/channel/Closing.kt

```diff
@@ -141,14 +141,15 @@ data class Closing(
                     }
                     watch is WatchEventConfirmed && watch.event is BITCOIN_ALTERNATIVE_COMMIT_TX_CONFIRMED -> when (val commitment = commitments.resolveCommitment(watch.tx)) {
                         is Commitment -> {
-                            logger.warning { "a commit tx for fundingTxIndex=${commitment.fundingTxIndex} fundingTxid=${commitment.fundingTxIndex} has been confirmed" }
+                            logger.warning { "a commit tx for fundingTxIndex=${commitment.fundingTxIndex} fundingTxId=${commitment.fundingTxId} has been confirmed" }
                             val commitments1 = commitments.copy(
                                 active = listOf(commitment),
                                 inactive = emptyList()
                             )
                             val newState = this@Closing.copy(commitments = commitments1)
+                            // This commitment may be revoked: we need to verify that its index matches our latest known index before overwriting our previous commitments.
                             when {
-                                watch.tx.txid == commitments1.latest.localCommit.publishableTxs.commitTx.tx.txid -> {
+                                watch.tx.txid == commitments1.latest.localCommit.publishableTxs.commitTx.tx.txid && commitments1.localCommitIndex == commitments.localCommitIndex -> {
```

Comment:
I think the previous code was correct: at this point there is nothing we can do, our local commitment (revoked or not) has been published and confirmed. We may as well try our luck and maybe we'll get away with it. With your addition, we will end up in the `handleRemoteSpentOther` and fail to recognize the tx.

Reply:
Right, if that happens our peer can spend all outputs anyway because they know our revocation secret, so we're screwed...